tst/EnergyPlus/unit/UnitaryHybridAirConditioner.unit.cc

```diff
@@ -0,0 +1,311 @@
+		std::string string = delimited_string(snippet);
+		ASSERT_FALSE(process_idf(string));
+
+		#define  TEMP_CURVE 0 
+		#define  W_CURVE 1
+		#define  POWER_CURVE 2
+
+		bool ErrorsFound(false);
+		GetZoneData(ErrorsFound);
+		EXPECT_FALSE(ErrorsFound);
+		// Initialize schedule values
+		DataGlobals::TimeStep = 1;
+		DataHVACGlobals::TimeStepSys = 1;
+		DataGlobals::NumOfTimeStepInHour = 1;
+		DataGlobals::MinutesPerTimeStep = 60;
+		DataEnvironment::Month = 1;
+		DataEnvironment::DayOfMonth = 21;
+		DataGlobals::HourOfDay = 1;
+		DataEnvironment::DSTIndicator = 0;
+		DataEnvironment::DayOfWeek = 2;
+		DataEnvironment::HolidayIndex = 0;
+		using General::JulianDay;
+		DataGlobals::WarmupFlag = false;
+
+		DataEnvironment::DayOfYear_Schedule = JulianDay(Month, DayOfMonth, 1);
+
+		ScheduleManager::UpdateScheduleValues();
+		Real64 SysMassFlow(0.0); // System supply mass flow rate [kg/s]
+		Real64 OAMassFlow(0.0); // OA mass flow rate [kg/s]
+								// Initialize zone areas and volumes - too many other things need to be set up to do these in the normal routines
+
+		DataHeatBalance::Zone(1).FloorArea = 232.26;
+		DataEnvironment::StdRhoAir = 1.225;
+		DataHeatBalance::ZoneIntGain.allocate(1);
+
+		SizingManager::GetOARequirements();
+		GetOAControllerInputs();
+		using DataZoneEquipment::CalcDesignSpecificationOutdoorAir;
+		bool UseOccSchFlag = true;
+		bool UseMinOASchFlag = true;
+		int OARequirementsPtr = 1;
+
+		//other stuff
+		CurveManager::GetCurveInputData(ErrorsFound);
+		EXPECT_FALSE(ErrorsFound);
+		using namespace EnergyPlus::DataEnvironment;
+
+		ProcessScheduleInput(); // read schedules
+
+		UpdateScheduleValues();
+		GetInputZoneHybridUnitaryAirConditioners(ErrorsFound);
+
+		GetOARequirements(); // get the OA requirements object
+		
+		EXPECT_FALSE(ErrorsFound);
+		InitZoneHybridUnitaryAirConditioners(1, 1);
+		Model * pZoneHybridUnitaryAirConditioner = HandleToHybridUnitaryAirConditioner(1);
+
+		Real64  Tosa, Tra, Wra, Wosa, RHosa, RHra, RequestedLoad, DesignMinVR, Requestedheating, RequestedCooling,Requested_Humidification, Requested_Dehumidification;
+		RHosa = 0;
+		std::string TimeDate;
+		int modenumber = 0;
+		Real64 MsaRatio, OSAF;
+		MsaRatio = OSAF = 1;
+		int lineN = 0;
+
+		Requestedheating = RequestedCooling= Requested_Humidification = Requested_Dehumidification = 0;
+
+		// There are a few input files that can be used in this unit test. Replace the file below with 
+		std::vector<std::string> lines = getAllLinesInFile2(configured_source_directory() + "/datasets/InputData_Mixed.csv"); 																							 //line input format: time, Outside Air Temperature(C),	Outside Air Relative Humidity(1 - 100 % ),	Outside Air Humidity Ratio(-)
+		Real64 NormalizationReference = 3.0176;
+		int Msa = NormalizationReference* MsaRatio;
+		int scaler = pZoneHybridUnitaryAirConditioner->ScalingFactor;
+		ofstream myfile;
+		myfile.open(configured_source_directory() + "/datasets/OutputData.csv");
```

Comment:
Really I'd prefer not to write to an output file specific to a unit test from within a unit test. This means you need to manually inspect the file to see if the outputs are what you expected or not. This goes against an automated testing framework and the advantages of unit tests and a CI system. You should really change the unit test to use the gtest assertions to test expected against actual.